vms/platformvm/block/builder/builder.go

```diff
@@ -92,28 +91,14 @@ func New(
 func (b *builder) BuildBlock(context.Context) (snowman.Block, error) {
 	b.Mempool.DisableAdding()
 	defer func() {
+		b.Mempool.RequestBuildBlock(false /*=emptyBlockPermitted*/)
```

Comment:
Why do we do this?

Reply:
We may want to build another block immediately after building a block. For example, the mempool is so full that there are more txs in it that can fit in a single block